spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.boot.actuate.info;
+
+import java.util.Map;
+
+/**
+ * information provider for the info endpoint
+ *
+ * @author Meang Akira Tanaka
+ */
+public interface InfoProvider {
+
+	/**
+	 * @return a collection of information
+	 */
+	Map<String, Object> provide();
```

Comment:
If we accept this PR, we'll have to break `InfoEndpoint`, yes.

I also think that the business of reading `info.*` should move to a default implementation of `InfoProvider`. You may want to review the health support as it has quite a lot of similarities with what you're trying to do. Thanks!